test/python/tf_unittest_runner.py

```diff
@@ -39,89 +42,147 @@ def main():
+    module_list: This is a list tensorflow test target names passed as an argument.
+    Example --list_tests=math_ops_test.R*
+    To get the list of tensorflow python test modules, query using bazel.
     bazel query 'kind(".*_test rule", //tensorflow/python/...)' --output label
+
+    regex_input: Regular expression input strings to filter and list tests. 
+    Few examples of accepted regex_input are:
+    math_ops_test
+    math_ops_test.DivNanTest
+    math_ops_test.DivNoNanTest.testBasic
+    math_ops_test.DivNoNanTest.*
+    math_ops_test.D*
+    math_ops_test.*
+    math_*_test
+    math_*_*_test
+    math*_test
     """
     loader = unittest.TestLoader()
-    module = __import__(test_module)
-    test_modules = loader.loadTestsFromModule(module)
     alltests = []
-    for test_class in test_modules:
-        alltests.append(([i.id() for i in test_class._tests]))
-    print('\n'.join((sorted(sum(alltests, [])))))
-
-
-def run_test(test_name, verbosity=2):
+    for test_module in module_list:
+        module = __import__(test_module)
+        if (module is None):
+            print("Enter a valid test name to run")
+        test_modules = loader.loadTestsFromModule(module)
+        for test_class in test_modules:
+            for i in test_class:
+                alltests.append(i.id())
+
+    if (re.search("\.", regex_input) is None):
+        return alltests
+    else:
+        test_name = (re.split("\*", regex_input))[0]
+        listtests = []
+        for test in alltests:
+            if test_name in test:
+                listtests.append(test)
+        return listtests
+
+
+def run_test(test_list, verbosity=2):
     """
     Runs a specific test suite or test case given with the fully qualified 
     test name and prints stdout.
 
     Args:
-    test_name: This is the test suite or the test case passed as an argument.
-    Example: --run_test=math_ops_test.AccumulateNTest        
+    test_list: This is the list of tests to run,filtered based on the 
+    regex_input passed as an argument.
+    Example: --run_test=math_ops_test.A*   
     verbosity: Python verbose logging is set to 2. You get the help string 
     of every test and the result.
     """
     loader = unittest.TestLoader()
-    tests = loader.loadTestsFromName(test_name)
-    test_result = unittest.TextTestRunner(verbosity=verbosity).run(tests)
-    testsRun = 0
-    tests_run = []
-    failures = []
-    errors = []
-    if test_result.wasSuccessful():
-        tests_run.append([testsRun, 1])
-        sys.exit()
-    elif test_result.errors:
-        errors.append([test_name, test_result.errors])
-        tests_run.append([testsRun, 0])
-    elif test_result.failures:
-        failures.append([test_name, test_result.failures])
-        tests_run.append([testsRun, -1])
+    for test in test_list:
+        tests = loader.loadTestsFromName(test)
+        test_result = unittest.TextTestRunner(verbosity=verbosity).run(tests)
```

Comment:
we do not do anything with the results?
We should print the test name and the result at least right?

Reply:
I am working on another PR to print the results and write the output to an XML file. So you will have test name, status and log.